qt/MainWindow.cc

```diff
@@ -1598,6 +1605,44 @@ void MainWindow::dropEvent(QDropEvent* event)
     }
 }
 
+bool MainWindow::event(QEvent* e)
+{
+    if (e->type() != QEvent::WindowActivate || !read_from_clipboard_)
+    {
+        return QMainWindow::event(e);
+    }
+
+    QClipboard const* clipboard = QGuiApplication::clipboard();
+    if (clipboard->text().trimmed().endsWith(QStringLiteral(".torrent"), Qt::CaseInsensitive) ||
+        clipboard->text().startsWith(QStringLiteral("magnet:"), Qt::CaseInsensitive))
```

Comment:
If we call `clipboard->text()` once instead of three times, it has some benefit:

- if Qt is building the same text each time, we avoid a little bit of Qt work
- we wouldn't need a `clipboard` local since its only use is to call text()
- `.startsWith()` on 1617 would benefit from the `trimmed()` on line 1616

```c++
auto const text = QGuiApplication::clipboard()->text().trimmed();
if (text.endsWith(...) || text.startsWith(...))
{
    for (auto const& entry : text.split(QLatin1Char('\n'))
   {
   }
}